src/gas_metering/mod.rs

```diff
@@ -352,6 +355,11 @@ fn determine_metered_blocks<R: Rules>(
         let instruction_cost = match rules.instruction_cost(instruction)? {
             InstructionCost::Fixed(c) => c,
             InstructionCost::Linear(base, cost_per) => {
+                // Enforce that cost per unit fits in 31 bits
+                if cost_per.get() >= 0x8000_0000 {
+                    return Err(anyhow!("cost per unit excedes the 0x80000000 limit"));
```

Comment:
Ah, I guess 32 bits plus 31 bits because _our_ gas is technically signed (although I'd like to remove that...). But yeah, this is reasonable.